apps/molgenis-components/src/components/forms/EditModalWizard.vue

```diff
@@ -66,22 +66,30 @@ export default {
   },
   mounted() {
     this.columnsSplitByHeadings = splitColumnsByHeadings(
-      this.tableMetaData.columns
+      this.tableMetaData.columns,
+      this.visibleColumns
     );
     this.$emit("setPageCount", this.columnsSplitByHeadings.length);
   },
   emits: ["setPageCount", "update:modelValue"],
 };
 
-function splitColumnsByHeadings(columns) {
+function splitColumnsByHeadings(columns, visibleColumns) {
   return columns.reduce((accum, column) => {
     if (column.columnType === "HEADING") {
       accum.push([{ name: column.name }]);
     } else {
       if (accum.length === 0) {
         accum.push([]);
       }
-      accum[accum.length - 1].push({ name: column.name });
+      if (
+        !visibleColumns ||
+        visibleColumns.find(
+          (visibleColumn) => column.name === visibleColumn.name
+        )
+      ) {
+        accum[accum.length - 1].push({ name: column.name });
+      }
```

Comment:
the function is called splitColumnsByHeadings and that what it does , it should not also filter columns with some visibility list. suggest make visibility a column prop.

Reply:
how is it resolved